### PR TITLE
Update to latest egui + use new Image api

### PR DESCRIPTION
**Pull request by jprochazk**
### What

- Update `egui` to point to latest commit on https://github.com/emilk/egui/pull/3328
- Remove `StaticImageCache`, as its a part of `egui` now
- Update icons to use `egui::Image`

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3311) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3311)
- [Docs preview](https://rerun.io/preview/5fd7b59260f843d1f2fab710b8f719b368a3f039/docs) 
- [Examples preview](https://rerun.io/preview/5fd7b59260f843d1f2fab710b8f719b368a3f039/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

**Comment by jprochazk** [comment]
Waiting for https://github.com/emilk/egui/pull/3328 to land on `master`

File: crates/re_ui/src/icons.rs [hunk 1]
```diff
@@ -10,6 +12,10 @@ impl Icon {
     pub const fn new(id: &'static str, png_bytes: &'static [u8]) -> Self {
         Self { id, png_bytes }
     }
+
+    pub fn as_image(&self) -> Image<'static> {
```
Comment (emilk) [inline crates/re_ui/src/icons.rs:16]: could be an `impl From<&Icon> for Image<'static>`

File: crates/re_viewport/src/viewport.rs [hunk 2]
```diff
@@ -552,9 +550,14 @@ impl TabWidget {
...
-            egui::widgets::Image::new(self.texture_id, self.icon_size).tint(self.text_color);
-        icon_image.paint_at(ui, self.icon_rect);
+        let icon_image = self
+            .icon
+            .as_image()
+            .fit_to_exact_size(self.icon_size)
+            .tint(self.text_color);
+        if let Ok(TexturePoll::Ready { texture }) = icon_image.load(ui) {
+            icon_image.paint_at(ui, self.icon_rect, &texture);
+        }
```
Comment (emilk) [inline crates/re_viewport/src/viewport.rs:560]: We do this pattern a lot now. We should make sure something like this works instead:

```suggestion
        icon_image.paint_at(ui, self.icon_rect);
```

I feel like everytime we as users see a naked `TexturePoll`, that's a failure in the egui interface

Comment (emilk) [reply crates/re_viewport/src/viewport.rs:560]: (we can save this for later)

**Pull request merged**